src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -770,6 +770,27 @@ protected function prepare_links( $id ) {
 			),
 		);
 
+		if ( post_type_supports( $this->post_type, 'revisions' ) ) {
+			$template = get_block_template( $id, $this->post_type );
+			if ( ! is_wp_error( $template ) && $template && $template->wp_id ) {
```

Comment:
```suggestion
			if ( ! is_wp_error( $template ) && ! empty( $template->wp_id ) ) {
```
A `! empty()` check can cover the second and third conditions.

Reply:
@costdev `get_block_template` can return as null. Before checking a property, we need to check it is not empty value.

Reply:
`! empty( $template->wp_id )` will cover both. See [this 3v4l](https://3v4l.org/7Q0O8).

Reply:
```suggestion
			if ( $template instanceof WP_Block_Template && $template->wp_id ) {
```

This feels cleaner. What do you think @costdev ?

Reply:
Looks good @spacedmonkey